pkg/v1/tkg/test/tkgctl/shared/e2e_tkrresolver.go

```diff
@@ -116,6 +124,32 @@ func E2ETKRResolverValidationForClusterCRUDSpec(context context.Context, inputGe
 		clusterConfigFile, err = framework.GetTempClusterConfigFile(input.E2EConfig.TkgClusterConfigPath, &options)
 		Expect(err).To(BeNil())
 
+		if input.OtherConfigs != nil {
```

Comment:
I don't think you need this

the format 
if isClusterClass, ok := input.OtherConfigs["clusterclass"];

Is meant just for that. If input.OtherConfigs is nil , ok evaluates to false... 
check this out, you can run this: 
```
type spec struct {
	OtherConfigs map[string]string
}
func main() {
	input := spec{}
	if isClusterClass, ok := input.OtherConfigs["clusterclass"]; ok && isClusterClass == "true" {
		println("what???? this is not possible")
	}
	if input.OtherConfigs == nil {
		println("This works ")
	}
}

Reply:
True, I can remove that check. Will do, so as to get rid of one unnecessary indentation level too.